pkg/identity/cache.go

```diff
@@ -114,23 +116,39 @@ func LookupIdentity(lbls labels.Labels) *Identity {
 	return NewIdentity(NumericIdentity(id), lbls)
 }
 
-// LookupReservedIdentity looks up a reserved identity by its labels and
+// LookupReservedIdentityByLabels looks up a reserved identity by its labels and
 // returns it if found. Returns nil if not found.
-func LookupReservedIdentity(lbls labels.Labels) *Identity {
-	// If there is only one label with the "reserved" source and a well-known
-	// key, return the well-known identity for that key.
-	if len(lbls) != 1 {
-		return nil
-	}
+func LookupReservedIdentityByLabels(lbls labels.Labels) *Identity {
```

Comment:
Can you add unit tests for this function?

Seems like before there was special casing to ensure that if there is exactly one label, it may return the exact numericidentity matching to a reserved identity, but now if there is multiple labels and any one of them matches any reserved identity then it will randomly return one of the reserved identities. I'm having trouble getting my head around how this is supposed to properly handle overlapping labels, or whether the caller is supposed to ensure that the set of labels provided doesn't include multiple reserved (or fixed) labels.

Reply:
@joestringer add comments+unit tests. But you are correct, the `reserved:label` should be the only label provided when doing the `LookupReservedIdentityByLabels`, if there are any other labels in given labels then we return nil. Do you think it should be a different logic?

Reply:
I think the valid combinations are:
* A single reserved label
* A single lbl.Key == labels.LabelKeyFixedIdentity with potentially additional identity labels but without any reserved labels
* If neither labels.LabelKeyFixedIdentity or a single reserved are specified nil must be returned
* In all other cases, we should log a warning.